### PR TITLE
Fix case sensitivity selection for find and rename ignorecase

### PR DESCRIPTION
**Pull request by scholtzan**
There was a small bug where the `case_sensitive` parameter in `find_status` was not parsed correctly.
Fixing this also made me rename `ignoreCase` to `caseSensitive` since it became quite confusing.

## Review Checklist

- [x] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code
- [x] I have updated comments / documentation related to the changes I made.
- [x] I have rebased my PR branch onto xi-mac/master.

### CODE REVIEW

File: Sources/XiEditor/Search.swift [hunk 1]
```diff
@@ -230,7 +230,7 @@ class SuplementaryFindViewController: NSViewController, NSSearchFieldDelegate, N
     override func validateMenuItem(_ menuItem: NSMenuItem) -> Bool {
         switch menuItem.tag {
         case ignoreCaseMenuTag:
```
Comment (nangtrongvuon) [inline Sources/XiEditor/Search.swift]: I'd probably rename this too while we're doing this.

File: Sources/XiEditor/Search.swift [hunk 2]
```diff
@@ -275,7 +275,7 @@ class SuplementaryFindViewController: NSViewController, NSSearchFieldDelegate, N
     }
 
     @IBAction func selectIgnoreCaseMenuAction(_ sender: NSMenuItem) {
```
Comment (nangtrongvuon) [inline Sources/XiEditor/Search.swift]: Likewise for this. Perhaps `selectCaseSensitiveMenuAction`?

File: Sources/XiEditor/Search.swift [hunk 3]
```diff
@@ -395,7 +395,7 @@ extension EditViewController {
             }
 
             if firstStatus.caseSensitive != nil {
-                query?.ignoreCase = !(statusQuery.caseSensitive != nil)
+                query?.caseSensitive = (statusQuery.caseSensitive != nil && statusQuery.caseSensitive!)
```
Comment (mmatoszko) [inline Sources/XiEditor/Search.swift]: I don't fully understand why is the `statusQuery.caseSensitive` negated here. Wasn't it just the case of the previous setting?

Comment (scholtzan) [reply Sources/XiEditor/Search.swift]: Yeah, the wording `ignoreCase` and `caseSensitive` was source for confusion before. Case sensitive is basically the opposite of ignoring case (if casing is ignored then it's _not_ sensitive to casing), that's why it had to be negated before.

**Pull request merged**